### PR TITLE
fix: ai watch-status reliability (health degradation, async safety, parallel probes)

### PR DESCRIPTION
**Pull request by jmagar**
## Summary

- **P1 fix**: `ai_watcher_process_start_time()` spawned `systemctl` via blocking I/O directly in an async fn — wrapped with `tokio::task::spawn_blocking` to avoid stalling the executor
- **P1 fix**: `health` degrades to `None` on DB failure instead of propagating the error — status command always returns `Ok` even during DB outages
- **D-Bus guard** unified: previously only applied to `journalctl` in `run_command`; now covers all `--user` commands in both `run_command` and `probe_command`
- **journalctl failure** surfaced as `journal_error: Option<String>` field — callers can distinguish "empty log" from "fetch failed"
- **5 parallel systemctl probes** via `tokio::join!` — worst-case latency drops from 150s to 30s
- **Warning logs** in `probe_systemctl` now log `args_owned` (includes `--user`) instead of bare args
- **4 new tests** covering main_pid parsing, health-field contract, journal error surfacing, exit code encoding fix

## Test plan

- [ ] All 856 unit tests pass (verified by pre-push hook)
- [ ] `syslog ai watch-status` returns `Ok` even when DB is unreachable
- [ ] `health: null` appears in JSON output during DB outage
- [ ] `journal_error` set when journalctl unavailable

## Open follow-ups

- `syslog-mcp-23xf` (P2): Update MCP schema to document `health: Option`
- `syslog-mcp-dliq` (P2): Add test that forces `health=None` branch
- `syslog-mcp-bl7t` (P3): Deduplicate D-Bus injection with OnceLock
- `syslog-mcp-s7te` (P3): Print `exec_main_start_timestamp` in CLI output
- `syslog-mcp-15h9` (P3): Document `journal_error`/`latest_journal` mutual exclusion

---
## Summary by cubic
Improves ai watch-status reliability and latency with parallel `systemctl` probes, async safety, and clear degradation paths when DB/journal are unavailable. Adds `syslog shell atuin-index` to import Atuin history, and updates JSON/CLI: `health` is optional with `health_error`/`journal_error`, and CLI shows `exec_main_start_timestamp`.

- **Bug Fixes**
  - Status always returns Ok; on DB failure `health=None` with `health_error`; journal failures set `journal_error` and keep logs empty (JSON omits absent fields).
  - Runs 5 `systemctl` probes in parallel; wraps blocking start-time call in `spawn_blocking`; improves probe handling/logs (accept non-empty stdout on non-zero exit; log full args).
  - Unifies D‑Bus env for all `--user` commands via `apply_dbus_env`; gates Unix-only tests; stabilizes compose tests with `syslog-mcp-data` and `SYSLOG_ENV_FILE` overrides; better spool error logs.

- **New Features**
  - `syslog shell atuin-index --path PATH [--json]` imports Atuin SQLite history with a resumable timestamp/id cursor.
  - Human output shows `exec_main_start_timestamp`; JSON adds `health_error` and `journal_error`; `health` is now optional.

<sup>Written for commit 610e1cb47e45a8bef3e109df95b0ecd87bf21a1d. Summary will update on new commits. <a href="https://cubic.dev/pr/jmagar/syslog-mcp/pull/55?utm_source=github">Review in cubic</a></sup>

## Summary by CodeRabbit

* **New Features**
  * Added Atuin shell-history import with a new local CLI command to run it.
  * Health status is now optional and shown as "unavailable" with an error detail when DB health cannot be read.
  * Journal probe now reports an explicit unavailable message with error detail when fetching fails.

* **Bug Fixes**
  * System probes and health checks degrade gracefully on partial failures instead of hard-erroring.

* **Chores**
  * Updated pre-commit hooks for linting and formatting checks.

* **Documentation**
  * Added a maintenance handoff session document.

* **Tests**
  * Expanded tests for graceful degradation, journal error reporting, PID/health parsing, and Atuin import behavior.

[Review Change Stack](https://app.coderabbit.ai/change-stack/jmagar/syslog-mcp/pull/55?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/app/watch_status_tests.rs [hunk 1]
```diff
@@ -6,14 +6,19 @@ use crate::app::{ServiceError, ServiceResult};
 use crate::config::StorageConfig;
 use crate::db::init_pool;
 
-fn exit_status(code: i32) -> std::process::ExitStatus {
+fn exit_code(code: u8) -> std::process::ExitStatus {
     use std::os::unix::process::ExitStatusExt;
-    std::process::ExitStatus::from_raw(code)
+    // from_raw encodes the wait(2) status word: exit code lives in the high byte.
+    std::process::ExitStatus::from_raw(i32::from(code) << 8)
 }
```
[the remark this answers is not in the document]
Comment (jmagar) [reply src/app/watch_status_tests.rs:15]: Fixed in aafcf7b — added  at the top of the file to gate the entire module on Unix targets.
